Source/KIS_Shared.cs

```diff
@@ -1354,7 +1354,7 @@ public enum MessageAction {
 
   /// <summary>Creates a new vessel from a single part.</summary>
   /// <remarks>Initially the part must belong to some vessel.</remarks>
-  static IEnumerator WaitAndMakeLonePart(Part newPart, OnPartReady onPartReady) {
+  static IEnumerator WaitAndMakeLonePart(Part newPart, Part fromPart, OnPartReady onPartReady) {
```

Comment:
You don't need an extra parameter. Initially, `newPart` belongs to the from vessel. Simply save the vessel (or the name) at the beginning of the method.

Reply:
Ah, ok. I just tested and that does indeed work nicely. Have a problem with ground attaching, though (probably something went wrong in the config patch).

Reply:
Turns out the attach problem was PEBCAK: I was using the drill instead of EL's mallet.